providers/azure/azure.go

```diff
@@ -241,54 +234,39 @@ func (b *Bucket) IsObjNotFoundErr(err error) bool {
 	if err == nil {
 		return false
 	}
-	storageErr := &azblob.StorageError{}
-	if ok := errors.As(err, &storageErr); !ok {
-		return false
-	}
-	return storageErr.ErrorCode == azblob.StorageErrorCodeBlobNotFound || storageErr.ErrorCode == azblob.StorageErrorCodeInvalidURI
+	return bloberror.HasCode(err, bloberror.BlobNotFound) || bloberror.HasCode(err, bloberror.InvalidURI)
 }
 
-func (b *Bucket) getBlobReader(ctx context.Context, name string, offset, length int64) (io.ReadCloser, error) {
-	level.Debug(b.logger).Log("msg", "getting blob", "blob", name, "offset", offset, "length", length)
+func (b *Bucket) getBlobReader(ctx context.Context, name string, httpRange blob.HTTPRange) (io.ReadCloser, error) {
+	level.Debug(b.logger).Log("msg", "getting blob", "blob", name, "offset", httpRange.Offset, "length", httpRange.Count)
 	if name == "" {
 		return nil, errors.New("blob name cannot be empty")
 	}
-	blobClient, err := b.containerClient.NewBlobClient(name)
-	if err != nil {
-		return nil, err
-	}
-	downloadOpt := &azblob.BlobDownloadOptions{
-		Offset: &offset,
-		Count:  &length,
+	blobClient := b.containerClient.NewBlobClient(name)
+	downloadOpt := &blob.DownloadStreamOptions{
+		Range: httpRange,
 	}
-	resp, err := blobClient.Download(ctx, downloadOpt)
+	resp, err := blobClient.DownloadStream(ctx, downloadOpt)
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot download blob, address: %s", blobClient.URL())
 	}
-
-	retryOpt := &azblob.RetryReaderOptions{
-		MaxRetryRequests: b.maxRetryRequests,
```

Comment:
Need to find where this should now be set.

Reply:
So it looks like the new SDK creates issues with reading streams with retries. Keeping this functionality would force reading all the data into memory, something that we have already tracked as a bug.

Looking at the AWS code it does not seem like it implements such logic for reading? If that is the case I suggest we move along with removing the retry logic here.